docs/agent-vs-aea.md

```diff
@@ -0,0 +1,238 @@
+AEAs are more than just agents. In this guide we show some of the differences in terms of code.
```

Comment:
@marcofavorito I have added this guide as there is an ongoing confusion by some users as to what an AEA and what an agent is. An agent is anything which perceives and acts based on the definition we use. Now this is reflected in our agent base class `Agent`. The benefit of this class is that it gives you access to the protocols and connections concepts without having to know anything about skills and the decision maker. Once the user reaches limits with this approach (as we did in the TAC), they can move to the `AEA`. Thoughts?

Reply:
We can then, in the intro section, guide users towards the approach which best fits their usecase.

Reply:
yes, it is a good outlook. Worth providing some examples and tests of "raw" agents.